src/NexusMods.MnemonicDB/Connection.cs

```diff
@@ -287,11 +287,28 @@ public IObservable<IChangeSet<Datom, DatomKey>> ObserveDatoms(SliceDescriptor de
                 var isMany = cache.IsCardinalityMany(datom.A);
                 changes.Add(new Change<Datom, DatomKey>(ChangeReason.Add, CreateKey(datom, datom.A, isMany), datom));
             }
-
+            
             if (changes.Count == 0)
-                return subject.StartWith(ChangeSet<Datom, DatomKey>.Empty);
-            return subject.StartWith(changes);
-        }
+                observer.OnNext(ChangeSet<Datom, DatomKey>.Empty);
+            else 
+                observer.OnNext(changes);
+            
+            
+            return Disposable.Create(() =>
+            {
+                _observersPendingDisposal.Enqueue(observer);
+            });
```

Comment:
Allocation of the closure can be eliminated:

```suggestion
            return Disposable.Create((_observersPendingDisposal, observer), static state =>
            {
                var (observersPendingDisposal, observer) = state;
                observersPendingDisposal.Enqueue(observer);
            });
```